source/ftrack_connect_pipeline_3dsmax/host/engine/asset_manager.py

```diff
@@ -92,75 +110,237 @@ def remove_asset(self, asset_info, options=None, plugin=None):
         result_data = {
             'plugin_name': plugin_name,
             'plugin_type': plugin_type,
-            'method': 'remove_asset',
+            'method': 'change_version',
             'status': status,
             'result': result,
             'execution_time': 0,
-            'message': message
+            'message': message,
         }
 
-        deleted_nodes = []
+        new_version_id = options['new_version_id']
 
-        ftrack_asset_object = self.get_ftrack_asset_object(asset_info)
+        self.asset_info = asset_info
+        dcc_object = self.DccObject(
+            from_id=asset_info[asset_const.ASSET_INFO_ID]
+        )
+        self.dcc_object = dcc_object
+
+        # It's an import, so change version with the main method
+        if (
+            self.dcc_object.get(asset_const.LOAD_MODE)
+            != modes_const.REFERENCE_MODE
+        ):
+            return super(MaxAssetManagerEngine, self).change_version(
+                asset_info=asset_info, options=options, plugin=plugin
+            )
 
+        # Unload the reference
+        unload_result = None
+        unload_status = None
         try:
-            # TODO: check if I have to get the object or not needed
-            # ftrack_object = ftrack_asset_object.ftrack_object.Object
-            deleted_nodes = max_utils.delete_all_children(ftrack_asset_object.ftrack_object)
-            status = constants.SUCCESS_STATUS
-        except Exception as error:
-            message = 'Could not delete nodes, error: {}'.format(error)
+            unload_status, unload_result = self.unload_asset(
+                asset_info=asset_info, options=None, plugin=None
+            )
+        except Exception as e:
+            unload_status = core_constants.ERROR_STATUS
+            self.logger.exception(e)
+            message = str(
+                "Error unloading asset with version id {} \n error: {} "
+                "\n asset_info: {}".format(
+                    asset_info[asset_const.VERSION_ID], e, asset_info
+                )
+            )
             self.logger.error(message)
-            status = constants.ERROR_STATUS
 
-        bool_status = constants.status_bool_mapping[status]
+        bool_status = core_constants.status_bool_mapping[unload_status]
         if not bool_status:
             end_time = time.time()
             total_time = end_time - start_time
 
             result_data['status'] = status
             result_data['result'] = result
             result_data['execution_time'] = total_time
-            result_data['message'] = message
+            result_data['message'] = result['message'] = message
 
             self._notify_client(plugin, result_data)
-            return status, result
 
+            return unload_status, unload_result
+
+        # Get reference Node
+        reference_node = None
+        # for obj in unload_result:
```

Comment:
Please clean up all commented code from maya

Reply:
Asset manager will be attended in a future PR